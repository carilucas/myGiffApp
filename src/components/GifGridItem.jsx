import { useEffect } from "react";


export const GifGridItem = ({title, id, img}) => {

   const titleTruncate = (title)=>{
      const dots = title.length > 30 ? '...':'';
      return title.substring(0,30) + dots;
   }

   const newTitle = titleTruncate(title)
   

  return (
    <div className="gridItem">
      <img src={ img } alt={ title } />
      <h4>{newTitle}</h4>
      <a href={img} target="_blank" rel="noopener noreferrer">Open Gif</a>
    </div>
  )
}
