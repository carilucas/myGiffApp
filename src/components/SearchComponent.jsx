import { useState } from "react";

import  PropTypes from 'prop-types';

export const SearchComponent = ({onChangeCategory}) => {

   const [searchValue, setSearchValue] = useState('');

   const handleInputChange = ( {target} )=>{
      setSearchValue( target.value );
   }

   const handleSubmit = (e)=>{
      e.preventDefault();
      if (searchValue.trim() <= 1) return;
      onChangeCategory( searchValue.trim() );
      setSearchValue('');
   }

  return (
   <>
   <p className="github">Github repository <a href="https://github.com/carilucas/myGiffApp" target="_blank" rel="noopener noreferrer">here</a></p>
   <form onSubmit={ handleSubmit } aria-label="form">
      <input 
         type="text" 
         name="" 
         id=""
         value={ searchValue }
         onChange ={ handleInputChange }
         placeholder="Search a gif"
      />
   </form>
   </>
  )
}


SearchComponent.propTypes = {
   onChangeCategory : PropTypes.func.isRequired
}