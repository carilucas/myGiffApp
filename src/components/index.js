export * from './SearchComponent';
export * from './GifGrid';
export * from './GifGridItem';