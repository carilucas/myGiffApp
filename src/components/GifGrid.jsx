
import { fetchGifs } from '../helpers/fetchGifs';
import { useFetch } from '../hooks/useFetch';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const GifGrid = ( {category} ) => {

   const { gifs, loading } = useFetch(category);
   
   return (
      <>
      <h2>{category.toUpperCase()}</h2>
      {
         loading && (<h1>......................LOADING...................</h1>)
      }
         <div className='grid'>
            {
               gifs.map( gif=> (
                  <GifGridItem key={gif.id} {...gif} />
               ))
            }
         </div>
      </>
  )
}

GifGrid.propTypes = {
   category: PropTypes.string.isRequired
}