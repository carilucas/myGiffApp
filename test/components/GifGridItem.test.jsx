import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

describe('Pruebas en <GifGridItem>', () => { 
   
   const title='titulo'; 
   const url='https://urldeimagen.com/'

   test('Debe mostrar el snapshot', () => { 

      const { container } = render( <GifGridItem title={title} img={url} /> );

      expect(container).toMatchSnapshot();

    })

    test('Debe mostrar la imagen con la url y el alt indicado', () => { 

      render( <GifGridItem title={title} img={url} /> );
      // screen.debug();

      // expect( screen.getByRole('img').alt).toBe(title);
      // expect( screen.getByRole('img').src).toBe(url);

      const {src,alt} = screen.getByRole('img')
      expect( src).toBe(url);
      expect( alt ).toBe(title);


     })

   test('Debe mostrar el tÃ­tulo en el componente', () => { 

      render( <GifGridItem title={title} img={url} /> );
      expect( screen.getByText( title)).toBeTruthy();


   }) 

 })