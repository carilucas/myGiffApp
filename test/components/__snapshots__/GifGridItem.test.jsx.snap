// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem> Debe mostrar el snapshot 1`] = `
<div>
  <div
    class="gridItem"
  >
    <img
      alt="titulo"
      src="https://urldeimagen.com/"
    />
    <h4>
      titulo
    </h4>
    <a
      class="openButton"
      href="https://urldeimagen.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Open Gif
    </a>
  </div>
</div>
`;
